README.md

```diff
@@ -1,75 +1,19 @@
-    │       |                                       and the pipeline to build and push image starts.
-    │       |
-    |       └── redeliver_container_image.md    <- If the tag exists and AICoE CI GitHub App (https://github.com/apps/aicoe-ci) is installed, the issue will retrigger the pipeline to build and
-    │                                               push image container image. NOTE: It should be used if the pipeline triggered with the {major|minor|patch}_release failed for any reason.
-    |
-    └── tox.ini                                 <- tox file with settings for running tox; see tox.readthedocs.io
+We fetch the CI test data from github and further store this to s3 storage. The main requirement here is to get test duration values. We can visualize the distribution of the test durations and based on the distribution type identified, we can find a point after which the test has a higher probability of failing.
 
---------
+## Feature Engineering
 
-<p><small>Project based on the <a target="_blank" href="https://drivendata.github.io/cookiecutter-data-science/">cookiecutter data science project template</a>. #cookiecutterdatascience</small></p>
+After fetching the data, we approximate the distributions of the test duration value and also check its goodness of fit for different tests. Based on the type of distribution identified, we can calculate the probability of the test failing.
+
+We fetch data for all the passing and failed tests and find the distribution type for the test duration. This tracks the time it took for a test to complete its execution. We can visualize the distribution of the test duration, based on the distribution type identified, we can find top two distributions based on betterment of fit. Probability density plots are used to understand data distribution for a continuous variable and likelihood (or probability) of obtaining a range of values that the continuous variable can assume. The area under the curve contains the probabilities for the test duration values. In a test duration probability distribution function, the area under the curve from 0 to a given value represents the probability that the test_duration is less than or equal to that value.
+
+## Model Training
+
+After performing initial data analysis and calulating the probability to fail, we predict the optimal stopping for a given test based on their test duration(runtimes). We find the best distribution(s) for the given test, and find an optimal stopping point for the test by finding the point where the probability of the test failing is greater than the probabilty of the test passing.
+
+Once this model is trained, it will use the Optimal stopping point value to further stop long running tests from consuming the resources after a certain point. This can help the engineering teams and managers better allocate their resources.
```

Comment:
```suggestion
Once we have calculated optimal stopping point values for tests, this can be used to stop long running tests from consuming resources after a certain point. This can help the engineering teams and managers better allocate their resources.
```

Is there a model training step here? Do we have a model or are we essentially calculating intersections between distribution curves? Should we rename the step to something better suited?

Reply:
Yes, we basically use statistical models to predict the distributions for the test durations and get the intersections. Maybe we should rename this. Maybe predict OSP, What do you think?

Reply:
Updated it to "Predicting the OSP" lmk wdyt?

Reply:
Awesome, updated.